sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SessionTest.java

```diff
@@ -127,6 +142,197 @@ public void sessionConsistency_ReadYourWrites(boolean isNameBased) {
+        }
+    }
+
+    @Test(groups = { "simple" }, timeOut = TIMEOUT, dataProvider = "sessionTestArgProvider")
+    public void sessionTokenNotRequired(boolean isNameBased) {
+        spyClient.readCollection(getCollectionLink(isNameBased), null).block();
+        spyClient.createDocument(getCollectionLink(isNameBased), newDocument(), null, false).block();
+
+        spyClient.clearCapturedRequests();
+        Document documentCreated = null;
+        RequestOptions requestOptions = new RequestOptions();
+        for (int i = 0; i < 10; i++) {
+            Document document = newDocument();
+            document.set("mypk", document.getId());
+            requestOptions.setPartitionKey(new PartitionKey(document.getId()));
+            documentCreated = spyClient.createDocument(getCollectionLink(isNameBased), document, requestOptions, false)
+                .block().getResource();
+        }
+        GlobalEndpointManager globalEndpointManager = ReflectionUtils.getGlobalEndpointManager(spyClient);
+        if(!globalEndpointManager.getLatestDatabaseAccount().getEnableMultipleWriteLocations()) {
+            // No session token set for write call
+            assertThat(getSessionTokensInRequests()).hasSize(0);
+        } else {
+            assertThat(getSessionTokensInRequests().size()).isGreaterThanOrEqualTo(1);
+            assertThat(getSessionTokensInRequests().get(0)).isNotEmpty();
+            assertThat(getSessionTokensInRequests().get(0)).doesNotContain(","); // making sure we have only one scope session token
+        }
+        spyClient.clearCapturedRequests();
+
+        // No session token set for EVENTUAL consistency
+        spyClient.clearCapturedRequests();
+        requestOptions.setConsistencyLevel(ConsistencyLevel.EVENTUAL);
+        spyClient.readDocument(getDocumentLink(documentCreated, isNameBased), requestOptions).block();
+        assertThat(getSessionTokensInRequests()).hasSize(0);
+
+        // No session token set for CONSISTENT_PREFIX consistency
+        spyClient.clearCapturedRequests();
+        requestOptions.setConsistencyLevel(ConsistencyLevel.CONSISTENT_PREFIX);
+        spyClient.readDocument(getDocumentLink(documentCreated, isNameBased), requestOptions).block();
+        assertThat(getSessionTokensInRequests()).hasSize(0);
+
+        // No session token set for BOUNDED_STALENESS consistency
+        spyClient.clearCapturedRequests();
+        requestOptions.setConsistencyLevel(ConsistencyLevel.BOUNDED_STALENESS);
+        spyClient.readDocument(getDocumentLink(documentCreated, isNameBased), requestOptions).block();
+        assertThat(getSessionTokensInRequests()).hasSize(0);
+
+        // No session token set for STRONG consistency
+        spyClient.clearCapturedRequests();
+        requestOptions.setConsistencyLevel(ConsistencyLevel.STRONG);
+        spyClient.readDocument(getDocumentLink(documentCreated, isNameBased), requestOptions).block();
```

Comment:
can we add one for the master resource validation?

Reply:
added check for collection read call